openslides/users/projector.py

```diff
@@ -9,14 +9,26 @@
 #            to be fast!
 
 
-def user(
-    element: Dict[str, Any], all_data: Dict[str, Dict[int, Dict[str, Any]]]
-) -> Dict[str, Any]:
+def user_slide(all_data: AllData, element: Dict[str, Any]) -> Dict[str, Any]:
     """
     User slide.
     """
     return {"error": "TODO"}
 
 
-def register_projector_elements() -> None:
-    register_projector_element("users/user", user)
+def get_user_short_name(all_data: AllData, user_id: int) -> str:
+    """
+    Returns the short name for an user_id.
+    """
+    user = all_data["users/user"][user_id]
+    name_parts: List[str] = []
+    for name_part in ("title", "first_name", "last_name"):
+        if user[name_part]:
+            name_parts.append(user[name_part])
+    if not name_part:
+        name_parts.append(user["username"])
+    return " ".join(name_parts)
```

Comment:
Why not returning an object as before?

Reply:
As discussed with @ostcar yesterday the server should send the final user string (in full name). The client can use it directly - without building a full name string.

Reply:
Users on projector should show always as full name (e.g. submitter, speaker, candidate, user slide). So the server can do the job to make the full name ready-to-use.